helix-term/src/ui/overlay.rs

```diff
@@ -0,0 +1,117 @@
+use crossterm::event::Event;
+use helix_core::Position;
+use helix_view::{
+    graphics::{CursorKind, Rect},
+    Editor,
+};
+use tui::buffer::Buffer;
+
+use crate::compositor::{Component, Context, EventResult};
+
+/// Contains a component placed in the center of the parent component
+#[derive(Debug)]
+pub struct Overlay<T> {
```

Comment:
Please separate this out, `Popup` with some calculation should be enough to replace this type altogether.

Reply:
(definitely avoid floats and use a percentage value between 0 and 100)

Reply:
It would be better if a type that can only guarantee the value range but I am not sure if we have something like that.

Reply:
@archseer I refactored the code again, it is now much simpler. Also, I'm not sure I understand what you mean with your first comment.